feedhenry/mobile-security-ios-template
# Add App Self-Defence

**Device Trust Checks**

- Jailbreak Detection
- Lock screen set
- Debug/debugger mode detection
- Emulator Detection
- Outdated OS Version Check

A trust score is also provided based on the detections.

There is less things to check for in comparison for Android. Checking for malicious/hooking applications were omitted as no jailbroken device was available for testing.

**iOS Snapshot Caching Protection**
Protection against iOS Snapshot Caching. When the app is put in the background, the app preview image is replaced with a blurred picture, so a screenshot of the current app screen is not saved to a world readable directory on iOS.

To verify:
- Open the app.
- Press the home button, and open another app. 
- Now open the app-switcher (double tap the home button)
- You should not see a preview of the app screen, just a blurred image.
- Click into the app, the blurred image will be gone on resume.

**Binary Protections**
I had a look at some buffer overflow protections:

- Automatic Reference Counting
- Address Space Layout Randomization
- Stack Smashing Protection

Its looks like for Swift, these are all enabled by default, unlike Objective C based apps, so no extra compiler flags needed to be added.
